Mage.Sets/src/mage/cards/e/Epicenter.java

```diff
@@ -0,0 +1,107 @@
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ *  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *  The views and conclusions contained in the software and documentation are those of the
+ *  authors and should not be interpreted as representing official policies, either expressed
+ *  or implied, of BetaSteward_at_googlemail.com.
+ */
+package mage.cards.e;
+
+import java.util.List;
+import java.util.UUID;
+import mage.abilities.Ability;
+import mage.abilities.condition.InvertCondition;
+import mage.abilities.condition.common.CardsInControllerGraveCondition;
+import mage.abilities.decorator.ConditionalOneShotEffect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.SacrificeEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.target.TargetPlayer;
+
+/**
+ *
+ * @author TheElk801
+ */
+public class Epicenter extends CardImpl {
+
+    public Epicenter(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{4}{R}");
+
+        // Target player sacrifices a land.
+        this.getSpellAbility().addEffect(new ConditionalOneShotEffect(
+                new SacrificeEffect(StaticFilters.FILTER_LAND, 1, "Target player"),
+                new InvertCondition(new CardsInControllerGraveCondition(7)),
+                "Target player sacrifices a land"));
+        // Threshold - Each player sacrifices all lands he or she controls instead if seven or more cards are in your graveyard.
+        this.getSpellAbility().addEffect(new ConditionalOneShotEffect(
+                new EpicenterEffect(),
+                new CardsInControllerGraveCondition(7),
+                "<br/><br/><i>Threshold</i> - Each player sacrifices all lands he or she controls instead if seven or more cards are in your graveyard."));
+
+        this.getSpellAbility().addTarget(new TargetPlayer());
+    }
+
+    public Epicenter(final Epicenter card) {
+        super(card);
+    }
+
+    @Override
+    public Epicenter copy() {
+        return new Epicenter(this);
+    }
+}
+
+class EpicenterEffect extends OneShotEffect {
+
+    EpicenterEffect() {
+        super(Outcome.DestroyPermanent);
+        staticText = "Each player sacrifices all lands he or she controls";
+    }
+
+    EpicenterEffect(final EpicenterEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        List<Permanent> permanents = game.getBattlefield().getActivePermanents(source.getControllerId(), game);
+        for (Permanent p : permanents) {
+            if (p.isLand()) {
+                p.sacrifice(source.getSourceId(), game);
+            }
```

Comment:
Yes, most of the time it will work.  Have you considered using iterations instead?  It is immune to the concurrent modification issue.